File: javascript/guess.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const getRandom = (min, max) => {
    return Math.floor((Math.random() * max) % max + min);
}

const min = 1
const max = process.argv[2] || 1000
const number = getRandom(min, max)

console.log(`I have selected a number between ${min} and ${max}.`)
console.log('Please guess my number.')

let guessCount = 0;

const askUser = () => {
    rl.question('Your guess: ', (n) => {
        guessCount ++;

        if (n == number) {
            console.log(`You guessed my number in ${guessCount} guesses.`)
            process.exit()
        } 
        else if (n < number) {
            console.log('Too low. Try again.');
            askUser();
        }
        else {
            console.log('Too high. Try again.')
            askUser();
        }
    })
}

askUser();

File: go/guess.go
package main

import (
	"fmt"
	"log"
	"math/rand"
	"os"
	"strconv"
)

func main() {
	min, max := 1, 1000

	if len(os.Args) > 1 {
		i, err := strconv.Atoi(os.Args[1])
		if err != nil {
			log.Panic(err)
		}
		max = i
	}

	number := rand.Intn(max-1) + min

	fmt.Printf("I have select a number between %d and %d.\n", min, max)
	fmt.Println("Pleast try to guess my number.")

	guessCount := 0

	for {
		var guess int
		fmt.Print("Your guess: ")
		_, err := fmt.Scanf("%d", &guess)

		if err != nil {
			fmt.Println("That doesn't look like a number. Try again.")
			continue
		}

		guessCount++

		if guess == number {
			fmt.Printf("You guessed my number in %d guesses!\n", guessCount)
			break
		} else if guess < number {
			fmt.Println("Too low. Try again.")
		} else {
			fmt.Println("Too high. Try again.")
		}
	}
}
